src/ci/docker/host-x86_64/x86_64-gnu-tools/checktools.sh

```diff
@@ -25,7 +25,7 @@ cat /tmp/toolstate/toolstates.json
 python3 "$X_PY" test --stage 2 check-tools
 python3 "$X_PY" test --stage 2 src/tools/clippy
 python3 "$X_PY" test --stage 2 src/tools/rustfmt
-python3 "$X_PY" test --stage 2 src/tools/miri
+MIRIFLAGS=-Zmiri-provenance-gc=1 python3 "$X_PY" test --stage 2 src/tools/miri
```

Comment:
Is this a temporary test or intended to stay?

Reply:
I really want the CI on this repo to be closer to the Miri repo. I'm iterating on what we can do to put this flag somewhere.

Reply:
Yeah, I'm not supposed to having the flag here. Though a brief comment would be good. And maybe a bit of space to separate this from the non-Miri tests which are in the same file.

Reply:
Note that this also gets run on Windows. In Miri we only do the provenance-gc thing on the Linux runner since it is the fastest. Might be good to do the same here.